### PR TITLE
raylib.com updates and qol additions

### PR DESCRIPTION
**Pull request by treblig-punisher**
**Problem:** 
-there was no way to quickly go back to the top of the page on mobile or desktop with a simple click.

**Solution:**
-added a go to top of page button that auto hides itself when at the top, and auto shows when the page is scrolled down a bit.

**Problem:** 
-The nav bar was cut off on mobile, and although it could be side scrolled, it was not fully visible.

**Solution:**
-Made the nav bar take the full available width when either on mobile or on android tablet device. 

**Problem:**
-On mobile, there was a forced width that ignored the screen size, causing some dom elements to [disrupt the page layout](https://imgur.com/a/x217135) 

**Problem:** 
-The search box had an issue where it would alter the layout whenever a search was being made. It also had no place holder text and no aria-label accessibility. It was also not responsive or centered.

**Solution:**
-Place holder text has been [added](https://imgur.com/a/ZNg9kh0). 
-Aria label has been added for text readers.
-The search box has been [styled](https://imgur.com/a/ZNg9kh0) to look more aesthetically pleasing & to provide the user with some visual cue.
-The search box now adapts to the screen size due to newly added responsiveness.
-The search results are now [underneath the search box](https://imgur.com/a/ZNg9kh0), keeping the flow of the layout consistent. 

**Problem:**
-The module boxes were not responsive and disrupted the page flow.

**Solution:**
-Added responsiveness to the modules container, and reworked how the module boxes are styled for better flexibility when styling.

**Problem:**
-The cheatsheet had no form of navigation outside of manual scrolling, resulting in a less than ideal user experience.

**Solution:**
-A modules menu has been added. Now users can click on any of the entries, and be auto scrolled to that section.
-This menu auto hides when print mode is enabled. 
-The menu is responsive.

**Problem**
-Navigation links were mission aria-labels.

**Solution**
-Added  aria-labels to navigation links so that screen readers can tell what they are.

**Problem**
-A navigation menu was missing for mobile.

**Solution**
-Mobile navigation menu implemented entirely

**A small note:**
I've added a js file, uiManager, that holds some logic regarding the go to top button. This will be used for future improvements where js is needed.

### CODE REVIEW

**Comment by raysan5** [comment]
@treblig-punisher Thanks for the review! I'm afraid some conflicts arise when reviewing some previous PRs. Please, could you review them?

I was testing the page and I noticed some things to review:

image

 - Is it possible to move the top-menu-icons to the right side, like it was previously. It looks quite overloaded on the left side.
 - Is it possible to review menu item names spacing? On current implementation it is spaced to fit the full page content and stay aligned.
 - There seem to be double-line spaces betweent paragraphs. I'd prefer to keep the simple spaces like current version.

Thanks again for the improvements!

**Comment by treblig-punisher** [comment]
> @treblig-punisher Thanks for the review! I'm afraid some conflicts arise when reviewing some previous PRs. Please, could you review them?
> 
> I was testing the page and I noticed some things to review:
> 
> image
> 
>     * Is it possible to move the top-menu-icons to the right side, like it was previously. It looks quite overloaded on the left side.
> 
>     * Is it possible to review menu item names spacing? On current implementation it is spaced to fit the full page content and stay aligned.
> 
>     * There seem to be double-line spaces betweent paragraphs. I'd prefer to keep the simple spaces like current version.
> 
> 
> Thanks again for the improvements!

Sure thing, I'll implement all changes and take a look at the conflict.

**Comment by treblig-punisher** [comment]
Hey Ray, I have a couple of questions, some updates, and a suggestion regarding the navbar.
I have already reverted back all the spacing related issues I introduced, and have adjusted the socials container so that it remains on the right and slides along with the the resizing of the window. I'll be adding those changes to my current branch shortly, but before I do that, there's this other topic regarding the menu names spacing.

I have previously made changes so that the page, along with the navbar react responsively to the window size. The original spacing for the menu items adds 20px of padding for each element, and this works fine if you have a forced size for the nav bar an d page, meaning this ignores reacting to changes to the window size, which keeps the navbar at that size while the view is smaller causing the current issues on mobile and desktop. Since I've modified how these two behave to accommodate for a more fluid and flexible design, this is what using the old padding-inline: 0px, 20px makes it look like:
image

The only compromise is having that scrollbar due to the current size of the page, which I am sure it's a big no no. I have a suggestion, and this includes your current suggestion regarding the padding to keep the spacing the same, but making the horizontal overflow hidden. Now, this suggestion requires having a hamburger menu ready, which I can get done and implemented in a day, so you might have to go ahead and not merge my changes yet. I'll deal with any conflicts after.

Now regarding the merge conflicts, is there a way i could solve them in vscode? most of these conflicts are present because I have things properly formatted to make it easier to read while working on it, but the same html code/css styles are long one liners in master. that's all I need to fix these and better spot new additions.

One more thing, would you be ok with having all the css in css files separate from the html files? It makes working with everything a bit more convenient. Let me know, and I'll work on that.

**Comment by treblig-punisher** [comment]
> @treblig-punisher Sincerely, I'd prefer to avoid it. Is not possible to slightly reduce padding, i.e. to 18px, to avoid that bar despite not getting the menu elements perfectly aligned?

Yes that's a great solution too. I'll implement that.

> I don't know, but they could be reformated if required. At the moment there are no coding-conventions defined for the web and that's something I'd like to do. I like to keep all the files and code very organized and following some conventions.

That's great to hear. For the most part, I'll work on a way to keep things better organized, and concise. so far everything is well structured, except for some of the css styling for some pages being on the html files themselves. 

> Yes, that's ok. Actually, as commented, the files structure and code conventions should be reviewed. Just one exception, I'd like to keep the cheatsheet as self-contained as possible, including the styling on the .html, at least for the moment.
 
 Alright, will do so. I'll keep that exception as needed 🫡

> Is this menu is for mobile only? Or will this menu also be for desktop mode? If only for mobile/responsive, that's ok to me.

That is correct, only mobile will have that.

One last thing, I will update my PR with all the changes, so you might have to give that a go, after, to see any if there are any new merge conflicts. I'll make sure to solve them.

**Comment by raysan5** [comment]
@treblig-punisher Nice! Thanks for the review! It seems there are still some conflicts that prevent merging...

**Comment by treblig-punisher** [comment]
> @treblig-punisher Nice! Thanks for the review! It seems there are still some conflicts that prevent merging...

I'll be taking a look at these soon. So far the good changes that wee needed have been added.

**Comment by treblig-punisher** [comment]
Alright Ray, I have resolved the conflicts in those two files. I added whatever was missing to mine, and removed anything that had been replaced. Let me know how it goes. I should work normally.

File: cls [hunk 1]
```diff
@@ -0,0 +1,2 @@
+* [32mgilbert-matos/raylib-website-updates[m
```
Comment (raysan5) [inline cls:1]: Is this file required?

File: games.html [hunk 2]
```diff
@@ -59,6 +64,27 @@
...
...85c-0,-11.526 9.358,-20.884 20.884,-20.884l199.898,-0c11.526,-0 20.884,9.358 20.884,20.884Z" style="fill:#727272;"/><path d="M171.441,35.354c0,11.526 -9.358,20.884 -20.884,20.884l-199.898,0c-11.526,0 -20.884,-9.358 -20.884,-20.884c-0,-11.527 9.358,-20.885 20.884,-20.885l199.898,-0c11.526,-0 20.884,9.358 20.884,20.885Z" style="fill:#727272;"/><path d="M171.441,92.275c0,11.526 -9.358,20.884 -20.884,20.884l-199.898,0c-11.526,0 -20.884,-9.358 -20.884,-20.884c-0,-11.527 9.358,-20.885 20.884,-20.885l199.898,-0c11.526,-0 20.884,9.358 20.884,20.885Z" style="fill:#727272;"/></g></g><g id="mobile-menu" serif:id="mobile menu"><rect x="-3095.22" y="-212.563" width="3529.17" height="6633.33" style="fill:#fff;"/><g id="x"><path d="M162.278,5.605c7.473,7.472 7.473,19.606 0,27.079l-129.594,129.594c-7.473,7.473 -19.607,7.473 -27.079,0c-7.473,-7.473 -7.473,-19.606 -0,-27.079l129.594,-129.594c7.473,-7.473 19.606,-7.473 27.079,-0Z" style="fill:#727272;"/><path d="M162.278,162.278c-7.473,7.473 -19.606,7.473 -27.079,0l-129.594,-129.594c-7.473,-7.473 -7.473,-19.607 -0,-27.079c7.472,-7.473 19.606,-7.473 27.079,-0l129.594,129.594c7.473,7.473 7.473,19.606 0,27.079Z" style="fill:#727272;"/></g></g></g></svg>
```
Comment (raysan5) [inline games.html:77]: Why all files include this SVG, couldn't it be replaced by the image logo directly?

File: path_to_raylib.html [hunk 3]
```diff
@@ -1 +1,197 @@
...ment.max=100*parseInt(t[4]),progressElement.hidden=!0,spinnerElement.hidden=!1):(progressElement.value=null,progressElement.max=null,progressElement.hidden=!0,e||(spinnerElement.style.display="none")),statusElement.innerHTML=e)}},totalDependencies:0,monitorRunDependencies:function(e){this.totalDependencies=Math.max(this.totalDependencies,e),Module.setStatus(e?"Preparing... ("+(this.totalDependencies-e)+"/"+this.totalDependencies+")":"All downloads complete.")}};Module.setStatus("Downloading..."),window.onerror=function(){Module.setStatus("Exception thrown, see JavaScript console"),spinnerElement.style.display="none",Module.setStatus=function(e){e&&Module.printErr("[post-exception status] "+e)}}</script><script>var audioBtn=document.querySelector("#btn-audio");const audioContexList=[];function toggleAudio(){var t=!1;audioContexList.forEach((e=>{"suspended"==e.state?(e.resume(),t=!0):"running"==e.state&&e.suspend()})),audioBtn.value=t?"🔇 SUSPEND":"🔈 RESUME"}self.AudioContext=new Proxy(self.AudioContext,{construct(t,e){const n=new t(...e);return audioContexList.push(n),"suspended"==n.state&&(audioBtn.value="🔈 RESUME"),n}})</script><script src=rtimeline.js async></script></body></html>
```
Comment (raysan5) [inline path_to_raylib.html:1]: This file intended to be minimized, why format was changed?

File: timeline.html [hunk 4]
```diff
@@ -1 +1,197 @@
...ment.max=100*parseInt(t[4]),progressElement.hidden=!0,spinnerElement.hidden=!1):(progressElement.value=null,progressElement.max=null,progressElement.hidden=!0,e||(spinnerElement.style.display="none")),statusElement.innerHTML=e)}},totalDependencies:0,monitorRunDependencies:function(e){this.totalDependencies=Math.max(this.totalDependencies,e),Module.setStatus(e?"Preparing... ("+(this.totalDependencies-e)+"/"+this.totalDependencies+")":"All downloads complete.")}};Module.setStatus("Downloading..."),window.onerror=function(){Module.setStatus("Exception thrown, see JavaScript console"),spinnerElement.style.display="none",Module.setStatus=function(e){e&&Module.printErr("[post-exception status] "+e)}}</script><script>var audioBtn=document.querySelector("#btn-audio");const audioContexList=[];function toggleAudio(){var t=!1;audioContexList.forEach((e=>{"suspended"==e.state?(e.resume(),t=!0):"running"==e.state&&e.suspend()})),audioBtn.value=t?"🔇 SUSPEND":"🔈 RESUME"}self.AudioContext=new Proxy(self.AudioContext,{construct(t,e){const n=new t(...e);return audioContexList.push(n),"suspended"==n.state&&(audioBtn.value="🔈 RESUME"),n}})</script><script src=rtimeline.js async></script></body></html>
```
Comment (raysan5) [inline timeline.html:1]: That file also intended to be minified...

File: uiManager.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
+const backToTopButton = document.querySelector(".back-to-top-arrow");
```
Comment (raysan5) [inline uiManager.js:1]: This file name does not seem to follow naming conventions for the web, probably it should be renamed to `ui_manager.js`

File: common/disable-arrow-scroll.js [hunk 6]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (raysan5) [inline common/disable-arrow-scroll.js:1]: This file name does not seem to follow naming conventions for the web, probably it should be renamed to `disable_arrow_scroll.js`

Is this file really required as a separate file or maybe the logic can be moved to `ui_manager.js`?

**Comment by raysan5** [comment]
@treblig-punisher Hi! Added some review comments!

File: common/disable-arrow-scroll.js (hunk 6 above)
Comment (treblig-punisher) [reply common/disable-arrow-scroll.js:1]: This is not my change, or file. This is one of the changes from someone else's commit I added it to fix the merge conflict. This came from your end when you shared the conflicts. I can move the logic to my file if you want me to though.

File: uiManager.js (hunk 5 above)
Comment (treblig-punisher) [reply uiManager.js:1]: My apologies. I'll fix that.

File: timeline.html (hunk 4 above)
Comment (treblig-punisher) [reply timeline.html:1]: Oh my apologies. This might've been the one file I formatted by accident when I was going through some open file tabs. I'll make sure to set it back to how it was.

File: games.html (hunk 2 above)
Comment (treblig-punisher) [reply games.html:77]: I'll refactor the entire thing, and add it in one place.

**Comment by treblig-punisher** [comment]
**Changes:**

-minified both path_to_raylib.html and timeline.html files. This was an unintentional format change that has now been reverted back to how it was.

- renamed uiManager.js to ui_manager.js and updated all references.

- deleted both hamburger svg, and x-icon svg. Their code has been moved to the ui_manager.js file.

- updated back-to-top-button's position on the page to prevent covering important page details when reaching the end of the page.

- added better mobile reposiveness to both **classic games and standard games buttons**. These will keep their full size when on desktop, and switch to a smaller width on smaller devices with less space.

- One change made to the social media links is the new support for aria-labels. Now screen readers can tell where the links lead instead of saying just "link." More aria-label changes in the near future to make raylib.com more accessible. 

- Renamed disable-arrow-scroll.js to disable_arrow_scroll.js. I didn't remove the file since whoever is making changes or using the file might need for future feature updates.

These are all the changes Ray. Let me know what you think. This should cover almost everything you suggested except for the thing I explained.

**Comment by raysan5** [comment]
@treblig-punisher I'm trying to review this PR but I'm afraid it's TOO big, could it be splitted in smaller PRs, please?

Also, I see many formatting changes that make it very difficult to see the real changes, could be the previous formatting maintained?

File: .vscode/settings.json [hunk 7]
```diff
@@ -0,0 +1,2 @@
+{
+}
```
Comment (raysan5) [inline .vscode/settings.json:2]: why is this empty file required?

File: cheatsheet/cheatsheet.html [hunk 8]
```diff
@@ -1,90 +1,371 @@
 <!DOCTYPE html>
 <html>
-    <head>
```
Comment (raysan5) [inline cheatsheet/cheatsheet.html:3]: Alignment changed, could it be kept the original alignment?

File: common/examples.css [hunk 9]
```diff
@@ -1,72 +1,364 @@
...
-#container .faudio a img, #container .faudio a { border: none; overflow: hidden; float: left; background-color: #ebddae; opacity: 1; }
-#container .faudio a:hover { border: 10px solid #d3b157; }
-#container .faudio a:hover img { margin: -10px; opacity: 0.3; }
-
-#container .fphysics a img, #container .fphysics a { border: none; overflow: hidden; float: left; background-color: #ebddae; opacity: 1; }
-#container .fphysics a:hover { border: 10px solid #d3b157; }
-#container .fphysics a:hover img { margin: -10px; opacity: 0.3; }
-
-#container .extext { position:absolute; top:140px; width:320px; text-align: center; display: none; }
+#container {
```
Comment (raysan5) [inline common/examples.css:1]: Could the CSS formatting be kept as previous solution, at least for the already defined properties?

File: common/games.css [hunk 10]
```diff
@@ -1,27 +1,132 @@
-#container .mix{ position: relative;  display: none; }
-#container .extext { top: 160px!important; width: 400px!important; }
```
Comment (raysan5) [inline common/games.css:2]: please, could be the formatting kept?

File: examples.html [hunk 11]
```diff
@@ -1,124 +1,156 @@
 <!DOCTYPE html>
 <html>
-    <head>
-        <meta charset="utf-8">
-        <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
 
-        <title>raylib - examples</title>
+<head>
```
Comment (raysan5) [inline examples.html:4]: Please, could be the spacing kept?

File: games.html [hunk 12]
```diff
@@ -33,6 +33,11 @@
     </head>
     
     <body>
+        <a href="#" class="back-to-top-arrow">
+            <svg xmlns="http://www.w3.org/2000/svg" width="23" height="23" viewBox="0 0 23 23" fill="none">
```
Comment (raysan5) [inline games.html:37]: Why is this SVG required?

File: index.html [hunk 13]
```diff
@@ -1,251 +1,466 @@
 <!DOCTYPE html>
 <html lang="en">
-    <head>
-        <meta charset="utf-8">
-        <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
-        <meta http-equiv="Content-Type" content="text/html; charset=utf-8">
 
-        <title>raylib | A simple and easy-to-use library to enjoy videogames programming</title>
+<head>
```
Comment (raysan5) [inline index.html:4]: Please, could be the aligned kept as current implementation?

**Comment by treblig-punisher** [comment]
> @treblig-punisher I'm trying to review this PR but I'm afraid it's TOO big, could it be splitted in smaller PRs, please?
> 
> Also, I see many formatting changes that make it very difficult to see the real changes, could be the previous formatting maintained?

I'll have to see how to split it to make it more manageable. 
Could you tell me what exactly changed regarding the format of the pages? what browser are you seeing these changes from?

File: .vscode/settings.json (hunk 7 above)
Comment (treblig-punisher) [reply .vscode/settings.json:2]: There was a PR beside mine that I had to merge with mine and fix some conflicts, so maybe that came with it, but I honestly don't remember ever making this file or its use. i can delete it if need be.

File: common/examples.css (hunk 9 above)
Comment (treblig-punisher) [reply common/examples.css:1]: Yes. I'll revert that back.

File: games.html (hunk 12 above)
Comment (treblig-punisher) [reply games.html:37]: this svg is for the button that you click to go back to the top of the page.

**Comment by raysan5** [comment]
> Could you tell me what exactly changed regarding the format of the pages?

Actually I was referring to code tabs spacing, as it changed it's difficult to see the what changed exactly for the PR.

It's a quite big PR and it's difficult to track review all changes in case something won't work as expected, so it's easier if it is split in smaller PRs, maybe by feature or modified pages. Code formatting is better to be maintained for the moment for changes readability.

**Comment by treblig-punisher** [comment]
> > Could you tell me what exactly changed regarding the format of the pages?
> 
> Actually I was referring to code tabs spacing, as it changed it's difficult to see the what changed exactly for the PR.
> 
> It's a quite big PR and it's difficult to track review all changes in case something won't work as expected, so it's easier if it is split in smaller PRs, maybe by feature or modified pages. Code formatting is better to be maintained for the moment for changes readability.

I totally understand. At the moment I am having a bit of a big issue given that the site has been updated again, and I am sure my changes will cause conflicts, since they are from 4 months ago, so here's what I propose:

I will clone the latest version of the repo, and try to implement my changes, and submit smaller PRs as you suggested. This is gonna take some time as I have to keep in mind what should change and what needs to remain minified.  The minified portions are meant to be like that for footprint purposes I am assuming, right?

The other thing regarding some of the changes is that I had to modify the nav bar elements to accommodate for the responsiveness of the site on tablets and mobile. 

Some the changes also include quality of life improvements for screen readers. This is aimed to help those with visual problems navigate the site with ease.

**About SVGs:**

I use svgs for some icons used by the updates I implemented like the go to top button, and the hamburger menu bar for mobile. I have these directly where I need to use them, in my case, I have them in the JS file that comes with the repo. This way I can just reuse it and not have to copy paste it in all pages they're used.

**On a final note:**

I am interested in also helping you create the new look for the site whenever you feel ready. You already shared a pretty good sketch of what you have in mind, so I'd love to further discuss this outside of github. Could be discord, or google meet, whichever works best for you.
